src/MetaMaskInpageProvider.ts

```diff
@@ -395,4 +395,54 @@ export default class MetaMaskInpageProvider extends BaseProvider {
       },
     );
   }
+
+  /**
+   * Upon receipt of a new chainId and networkVersion, emits corresponding
+   * events and sets relevant public state.
+   * Does nothing if neither the chainId nor the networkVersion are different
+   * from existing values.
+   *
+   * @emits MetamaskInpageProvider#chainChanged
```

Comment:
We should add `networkChanged` here as well:
```suggestion
   * @emits MetamaskInpageProvider#chainChanged
   * @emits MetamaskInpageProvider#networkChanged
```